app/components/publish-actions.tsx

```diff
@@ -298,68 +199,193 @@ const Share = ({ configKey, locale }: PublishActionProps) => {
         );
       }}
     >
-      <Box m={4} sx={{ "& > * + *": { mt: 4 } }}>
-        <Flex
-          sx={{
-            justifyContent: "space-between",
-            alignItems: "center",
-            mb: 4,
-          }}
-        >
-          <Typography component="div" variant="body1" color="grey.700">
-            <Trans id="publication.popup.share">Share</Trans>:
-          </Typography>
-          <Flex color="primary">
-            <IconLink
-              iconName="facebook"
-              title={i18n._(
-                t({
-                  id: "publication.share.linktitle.facebook",
-                  message: `Share on Facebook`,
-                })
-              )}
-              href={`https://www.facebook.com/sharer/sharer.php?u=${shareUrl}`}
-            ></IconLink>
-            <IconLink
-              iconName="twitter"
-              title={i18n._(
-                t({
-                  id: "publication.share.linktitle.twitter",
-                  message: `Share on Twitter`,
-                })
-              )}
-              href={`https://twitter.com/intent/tweet?url=${shareUrl}&via=bafuCH`}
-            ></IconLink>
-            <IconLink
-              iconName="mail"
-              title={i18n._(
-                t({
-                  id: "publication.share.linktitle.mail",
-                  message: `Share via email`,
-                })
-              )}
-              href={`mailto:?subject=${i18n._(
-                t({
-                  id: "publication.share.mail.subject",
-                  message: `visualize.admin.ch`,
-                })
-              )}&body=${i18n._(
-                t({
-                  id: "publication.share.mail.body",
-                  message: `Here is a link to a visualization I created on visualize.admin.ch`,
-                })
-              )}: ${shareUrl}`}
-            ></IconLink>
-          </Flex>
+      <ShareContent configKey={configKey} locale={locale} />
+    </TriggeredPopover>
+  );
+};
+
+type EmnbedContentProps = {
+  iframeHeight: number;
+} & Omit<PublishActionProps, "chartWrapperRef">;
+
+export const EmbedContent = ({
+  locale,
+  configKey,
+  iframeHeight,
+}: EmnbedContentProps) => {
+  const [embedUrl, setEmbedUrl] = useState("");
```

Comment:
This is a personal opinion, but I think it was a bit more clear to have them inside one component, without a need to jump from `Embed` to `EmbedContent` – with the current approach, we need to have additional component, component props (with omitting some of them) and to pass them down, while the `Embed` and `Share` components are very bare-bone now and I think `Content` components are only used one inside their parents.

But it's okay like this, maybe there was some other reason I missed to separate them 👍

Reply:
Yeah, sorry for the confusion I'm currently working on issue #1758 which doesnt need the buttons, inside the user profile it's meant to have the same design but opened by a `MenuActionItem`

Reply:
I see, thanks for clarifying 👍